### ISSUE TITLE
Flaky E2E tests due to istio-prometheus-test not starting

### ISSUE THREAD

**Issue by hklai** [istio/istio#11023]
E.g. 
https://gubernator.k8s.io/build/istio-prow/logs/e2e-simpleTests/15836
https://gubernator.k8s.io/build/istio-prow/logs/e2e-simpleTests/15328

`FAILED: istio-prometheus-test, run `kubectl logs istio-prometheus-test --namespace istio-system` for more info`

```
istio-citadel-5fd64f4d67-cq2n7            1/1       Running     0          2m
istio-egressgateway-6c8974cf49-g9dgh      1/1       Running     0          2m
istio-galley-5b86bbd9c7-pjvlf             1/1       Running     0          2m
istio-ingressgateway-cf4fd8bcb-45xh2      1/1       Running     0          2m
istio-pilot-7cc567b95d-gwsh5              2/2       Running     0          2m
istio-policy-668c9f7d76-7crzs             2/2       Running     3          2m
istio-prometheus-test                     0/1       Error       0          2m
istio-security-test                       0/1       Completed   0          2s
istio-sidecar-injector-76756c8bfc-f8dwz   1/1       Running     0          2m
istio-telemetry-786964fc5d-rfx54          2/2       Running     3          2m
prometheus-7549ccd956-6dwv7               1/1       Running     0          2m
```

**Comment by douglas-reid** [comment]
Can we just remove this test?  I'm not sure why it was added nor what purpose it serves.

**Comment by hklai** [comment]
@douglas-reid This is happening to e2e-simpleTests What are you suggesting to remove?

**Comment by douglas-reid** [comment]
this is coming from some `test/` directory under the helm artifacts from the `prometheus` subchart.  `e2e-simple` has apparently been expanded to do helm validation. I'm wondering if we need any validation of prometheus starting, seeing as we have other tests already that depend on it... and because I don't know how to debug helm test artifacts.

**Comment by hklai** [comment]
@douglas-reid are you talking about https://github.com/istio/istio/blob/master/install/kubernetes/helm/subcharts/prometheus/templates/tests/test-prometheus-connection.yaml?

It was added byhttps://github.com/istio/istio/pull/10623.
cc @morvencao and @linsun

**Comment by sdake** [comment]
Fixed and merged in release-1.1:
https://github.com/istio/istio/pull/11360/files

**Comment by ozevren** [comment]
@sdake, can we close this bug now?
